packages/agents-core/src/agent.ts

```diff
@@ -1,4 +1,4 @@
-import type { ZodObject } from 'zod/v3';
```

Comment:
Although `zod/index` refers to `zod/v3/external.js` within zod package, having `zod/v3` inside this Agents SDK causes TS compilation errors. Thus, simply having references to 'zod' is the right way to rely on zod on the SDK side.